### ISSUE TITLE
Question on writing deriver for class

### ISSUE THREAD

**Issue by Wenke-D** [ocaml-ppx/ppxlib#509]
It appears impossible to get the attache class AST using **context free transformation**.

```OCaml
class apple = object
   ...
end [@@inspect]
```
in the ppxlib, `Deriving.add` do not have the optional arg to get `class_declaration list`.

Did I miss something or it is indeed impossible.

Thanks !

**Comment by Wenke-D** [comment]
The objective is to inspect the methods and fields the class, then generate some functions outside.

**Comment by NathanReb** [comment]
Yes indeed it seems it's not supported. Off the top of my head I can't think of a reason why we wouldn't add support for this!
